src/charts/brush.js

```diff
@@ -51,20 +57,147 @@ export default function Brush (_container) {
         .on("start", handleBrushStart)
         .on("brush", handleBrushMove)
         .on("end", handleBrushEnd)
+
+      /* okay, this is a little esoteric - in an ideal world, we'd probably have a separate "Zoom"
+         component that would live alongside our "Brush" component here and handle the zooming. the
+         problem is that d3.zoom swallows all mousemove events, which prevents us from brushing on the chart.
+         (and you'll recall that we need to brush on the chart to set a filter, in addition to zooming in and out).
+
+         So the solution is twofold - first of all, we do the zooming in here, and secondly we attach the zoom
+         event to the <g> tag brush container. d3.brush adds on a few elements that directly handle the interaction,
+         so we let them absorb the mousemove events for the brushing and bubble up the wheel events to the zoom
+         in the container tag.
+
+      */
+      cache.zoom = d3.zoom()
+        .on("zoom", handleZoom)
+
+      cache.root.call(cache.zoom)
+
     }
   }
 
   function buildBrush () {
     cache.brush.extent([[0, 0], [config.markPanelWidth, config.chartHeight]])
 
     cache.chartBrush = cache.root.call(cache.brush)
+    cache.root.call(cache.zoom)
 
     cache.chartBrush.selectAll(".brush-rect")
       .attr("height", config.chartHeight)
 
     setBrush()
   }
 
+  function handleZoom () {
+
+    // this is just an inherently complex (and yet self-contained!) function. Sorry.
```

Comment:
Thank you for the comments. Makes my job as reviewer much easier, and helps anybody else working on this.